gravitee-apim-rest-api/gravitee-apim-rest-api-management-v4/gravitee-apim-rest-api-management-v4-rest/src/test/java/io/gravitee/rest/api/management/v4/rest/resource/connector/EntrypointsResourceTest.java

```diff
@@ -60,29 +57,103 @@ public void init() {
+        when(entrypointConnectorPluginService.findAll())
+            .thenReturn(
+                List
+                    .of(getConnectorPluginEntity("id-1"), getConnectorPluginEntity("id-2"), getConnectorPluginEntity("id-3"))
+                    .stream()
+                    .collect(Collectors.toCollection(LinkedHashSet::new))
+            );
 
         final Response response = rootTarget().request().get();
         assertEquals(HttpStatusCode.OK_200, response.getStatus());
-        final List<Map<String, String>> pluginEntities = response.readEntity(List.class);
-        assertEquals(1, pluginEntities.size());
-        Map<String, String> pluginEntity = pluginEntities.get(0);
-        assertEquals("id", pluginEntity.get("id"));
-        assertEquals("name", pluginEntity.get("name"));
-        assertEquals("1.0", pluginEntity.get("version"));
-        assertEquals(ApiType.MESSAGE.getLabel(), pluginEntity.get("supportedApiType"));
-        assertEquals(ListenerType.HTTP.getLabel(), pluginEntity.get("supportedListenerType"));
-        ArrayList<String> arrayList = new ArrayList<>();
-        arrayList.add(ConnectorMode.SUBSCRIBE.getLabel());
-        assertEquals(arrayList, pluginEntity.get("supportedModes"));
+
+        // Check response content
+        final EntrypointsResponse entrypointsResponse = response.readEntity(EntrypointsResponse.class);
+        assertNotNull(entrypointsResponse.getData());
+        assertNotNull(entrypointsResponse.getPagination());
+        assertNotNull(entrypointsResponse.getLinks());
+
+        // Check data
+        List<ConnectorPlugin> data = entrypointsResponse.getData();
+        assertEquals(3, data.size());
+        ConnectorPlugin pluginEntity = data.get(0);
+        assertEquals("id-1", pluginEntity.getId());
+        assertEquals("name", pluginEntity.getName());
+        assertEquals("1.0", pluginEntity.getVersion());
+        assertEquals(ApiType.MESSAGE.getLabel(), pluginEntity.getSupportedApiType().getValue());
+        assertEquals(ListenerType.HTTP.getLabel(), pluginEntity.getSupportedListenerType().getValue());
+        assertEquals(1, pluginEntity.getSupportedModes().size());
+        assertEquals(ConnectorMode.SUBSCRIBE.getLabel(), pluginEntity.getSupportedModes().iterator().next().getValue());
+
+        // Check pagination
+        Pagination pagination = entrypointsResponse.getPagination();
+        assertEquals(1, pagination.getCurrentPage());
+        assertEquals(10, pagination.getNumberOfItemPerPage());
+        assertEquals(3, pagination.getTotalNumberOfItems());
+        assertEquals(1, pagination.getTotalNumberOfPages());
+
+        // Check links
+        Links links = entrypointsResponse.getLinks();
+        assertTrue(links.getSelf().endsWith("/plugins/entrypoints/"));
+        assertNull(links.getFirst());
+        assertNull(links.getPrevious());
+        assertNull(links.getNext());
+        assertNull(links.getLast());
```

Comment:
Do you think we should send nothing if there is only one page?
I would suggest this:
```suggestion
        assertTrue(links.getFirst().endsWith("/plugins/entrypoints?page=1&perPage=10);
        assertNull(links.getPrevious());
        assertNull(links.getNext());
        assertTrue(links.getLast().endsWith("/plugins/entrypoints?page=1&perPage=10);
```